### PR TITLE
[3-15] JWT authorization

### PR DESCRIPTION
**Pull request by HenadzV**
1. Added JWT authentication/authorization (auth.service)
2. After server response store JWT and JWT expiration date in LS
3. Added http interceptor, which add JWT to all http requests (for authorization)
4. Added logic for logout button - after logout  clear LS and redirect to login page

image

### CODE REVIEW

**Review by vstrabko — APPROVED** [review]
Nice work!

**Review by HrebenYury — APPROVED** [review]

**Review by YanaBr — APPROVED** [review]

File: exalearn-ui/src/app/services/auth.service.ts [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+		this.router.navigate(['/login']);
+	}
+
+	isAuthenticated(): boolean {
+		return !!this.token;
+	}
+
+	private setToken(response: serverAuthResponse | null): void {
+		if (response) {
+			const expDate = new Date(new Date().getTime() + 3600 * 3 * 1000);
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/auth.service.ts]: I think you can move expDuration constants to the readonly variable of the service

**Review by YanaBr — APPROVED** [review]

File: exalearn-ui/src/app/services/auth.service.ts (hunk 1 above)
Comment (HenadzV) [reply exalearn-ui/src/app/services/auth.service.ts]: Done

**Comment by KotdenGit** [comment]
Branch naming is similar to commits. Please pay attention to this another time.

**Pull request merged**